test/e2e/config/docker.yaml

```diff
@@ -6,6 +6,7 @@
 # - docker
 
 # For creating local dev images run make docker-build-e2e from the main CAPI repository
+# File format: https://pkg.go.dev/sigs.k8s.io/cluster-api/test/framework/clusterctl#E2EConfig
```

Comment:
Good idea - this can definitely be confusing for users. 

Could you move this to the top of the file - and add something like: 
```
# This is a file for CAPI end-to-end test configuration. The format of this file follows https://pkg.go.dev/sigs.k8s.io/cluster-api/test/framework/clusterctl#E2EConfig
```